cachito/workers/pkg_managers/gomod.py

```diff
@@ -465,13 +491,39 @@ def go_list_deps(pattern: Literal["./...", "all"]) -> Iterator[GoPackage]:
             _vet_local_file_dep_paths(package["pkg_deps"], app_source_path, git_dir_path)
             _set_full_local_dep_relpaths(package["pkg_deps"], main_module_deps)
 
+        main_module_dict = {
+            "type": "gomod",
+            "name": local_modules.main.path,
+            "version": local_modules.main.version,
+        }
+
         return {
-            "module": main_module,
+            "module": main_module_dict,
             "module_deps": main_module_deps,
             "packages": main_packages,
         }
 
 
+def _set_local_modules_versions(
```

Comment:
I don't like the idea of updating the object attributes in place, but I also didn't want to do a deep dive into refactoring `get_golang_version`. This function takes a lot of attributes, and anything I wrote in the LocalModules class would bring all this crap together :confused: